plugins/outputs/mqtt/mqtt.go

```diff
@@ -58,6 +61,16 @@ func (*MQTT) SampleConfig() string {
 	return sampleConfig
 }
 
+func (m *MQTT) Init() error {
+	m.template = template.Must(template.New("topic_name").Parse(m.Topic))
+	for _, p := range strings.Split(m.Topic, "/") {
+		if strings.ContainsAny(p, "#+") {
+			return fmt.Errorf("found forbidden character %s in the topic name %s", p, m.Topic)
+		}
+	}
```

Comment:
Please don't use `Must` as this will lead to a panic of telegraf for invalid patterns.
```suggestion
	for _, p := range strings.Split(m.Topic, "/") {
		if strings.ContainsAny(p, "#+") {
			return fmt.Errorf("found forbidden character %q in the topic %q", p, m.Topic)
		}
	}
	tmpl, err := template.New("topic_name").Parse(m.Topic)
	if err != nil {
		return fmt.Errorf("invalid topic template %q: %w", m.Topic, err)
	}
	m.template = tmpl
```

Reply:
🤔 I had done that intentional as a means to avoid initiation of telegraf if the template wasn't possible to be parsed.
@srebhan what is your expectation here if the topic isn't parseable by the template? what should happen then?